crates/state/src/wl_state.rs

```diff
@@ -651,6 +651,29 @@ where
         }
     }
 
+    /// Borrow in-memory state and DB handle with a mutable temporary write-log.
+    ///
+    /// # Safety
+    ///
+    /// The lifetime of borrows is unsafely extended to `'static` to allow usage
+    /// in node's `dry_run_tx`, which needs a static lifetime to be able to call
+    /// protocol's API that is generic over the state with a bound `S: 'static +
+    /// State` with a mutable reference to this struct.
+    /// Because the lifetime of `S` is invariant w.r.t. `&mut S`
+    /// (<https://doc.rust-lang.org/nomicon/subtyping.html>) we are faking a
+    /// static lifetime of `S` for `TempWlState`. This should be safe as neither
+    /// `db` nor `in_mem` are actually mutable, only the `write_log` which is
+    /// owned by this struct.
```

Comment:
I think this is the explanation that should go under those `unsafe` blocks. here we should only write something like "the caller must guarantee that ..."

Reply:
improved this in a04ad35055a9ac240b260008b726740400f6dad9